### PR TITLE
Special filesystem handling for PASE

### PR DESCRIPTION
**Pull request by aremmell**
* Reuse $PATH resolution of binary name from OpenBSD
* Only use fstatat to detect if a file exists when relative to CWD

### CODE REVIEW

**Review by johnsonjh — DISMISSED** [review]

File: include/sir/platform.h [hunk 1]
```diff
@@ -450,13 +452,16 @@ _set_thread_local_invalid_parameter_handler(
 #   include <stdatomic.h>
 #  endif
 #  if defined(SIR_SYSLOG_ENABLED)
-#   include <syslog.h>
+#   if !defined(__cplusplus)
```
Comment (aremmell) [inline include/sir/platform.h:455]: Why if !defined C++?

File: src/sirfilesystem.c [hunk 2]
```diff
@@ -77,13 +77,26 @@ bool _sir_pathgetstat(const char* restrict path, struct stat* restrict st, sir_r
...
             return _sir_handleerr(errno);
         }
-
         stat_ret = fstatat(fd, path, st, AT_SYMLINK_NOFOLLOW);
+# else
+        // HACKHACK: fstatat does not work properly for any fd other than AT_FDCWD.
+        SIR_UNUSED(open_flags);
+        int fd = AT_FDCWD;
+        if (rel_to == SIR_PATH_REL_TO_CWD) {
+            /* cppcheck-suppress invalidFunctionArg */
```
Comment (aremmell) [inline src/sirfilesystem.c:92]: Where is there an invalid arg error from cppcheck?

File: src/sirfilesystem.c [hunk 3]
```diff
@@ -457,21 +459,29 @@ int _sir_aixself(char* buffer, size_t* size) {
     (void)snprintf(pp, sizeof(pp), "/proc/%llu/psinfo", (unsigned long long)_sir_getpid());
 
     fd = open(pp, O_RDONLY);
-    if (fd < 0)
+    if (fd < 0) {
+    _sir_selflog("failed to open %s; errno: %d", pp, errno);
```
Comment (aremmell) [inline src/sirfilesystem.c:463]: whoops, I didn't mean to leave this selflog in.

File: src/sirfilesystem.c [hunk 4]
```diff
@@ -457,21 +459,29 @@ int _sir_aixself(char* buffer, size_t* size) {
...
         return -1;
+    }
 
     res = read(fd, &ps, sizeof(ps));
     close(fd);
-    if (res < 0)
+    if (res < 0) {
+        _sir_selflog("failed to read %d; errno: %d", fd, errno);
```
Comment (aremmell) [inline src/sirfilesystem.c:470]: or this one

File: src/sirfilesystem.c [hunk 5]
```diff
@@ -457,21 +459,29 @@ int _sir_aixself(char* buffer, size_t* size) {
...
         return -1;
+    }
 
-    if (ps.pr_argv == 0)
+    if (ps.pr_argv == 0) {
+        _sir_selflog("error: ps.pr_argv == 0");
```
Comment (aremmell) [inline src/sirfilesystem.c:475]: all of them

File: include/sir/platform.h (hunk 1 above)
Comment (johnsonjh) [reply include/sir/platform.h:455]: On some platforms including the syslog C header directly within C++ is unsupported.  PASE is one.  I should make that just for PASE?

File: src/sirfilesystem.c (hunk 2 above)
Comment (johnsonjh) [reply src/sirfilesystem.c:92]: I'll remove it to allow it to fail.

**Review by johnsonjh — APPROVED** [review]

**Review by johnsonjh — APPROVED** [review]

**Review by johnsonjh — APPROVED** [review]

**Pull request merged**